test/controllers/refresh_tokens_controller_test.rb

```diff
@@ -0,0 +1,96 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+class RefreshTokensControllerTest < ActionDispatch::IntegrationTest
+  test 'when attempting to renew access token without an access cookie' do
+    post token_refresh_url
+
+    assert_response :unauthorized
+  end
+
+  test 'when renewing an access token one second before access-expiry' do
+    Timecop.freeze
+
+    token(users(:sterling_archer))
+
+    Timecop.travel((JWTSessions.access_exp_time - 1).seconds.from_now)
+
+    post token_refresh_url, headers: @headers
+
+    # TODO: sometimes this assertion fails; if you re-run the test it might work.  Timecop must be fucked?
```

Comment:
I assume this is because sometimes all this code executes in less than a second and sometimes it does not, I bet it always works if you did something like 15 seconds.  I think that would be a small enough window, would require changing the test sentence.